### DISCUSSION TITLE
Hi few questions I am not sure I understood correctly

### DISCUSSION THREAD

**Discussion by boynet** [netflie/whatsapp-cloud-api#9] [Q&A]
first of all, looks really good. thanks a lot for the library

1.  how would a $component_header will look like when your header is an image?
2. when sending images is it enough to put use `new LinkID('http(s)://image-url');` ? as from the api guide it seems we first need to upload the media to their server and than attaching the returned id
3. any plan to support for Interactive object message?

**Comment by boynet** [comment]
1. this work(maybe Its better to use LinkId instead):
```
$component_header = [
            [
                'type'=>'image',
                'image' => ['link'=>'http://example.com/example.jpg']
            ]
        ];
```

**Reply by aalbarca** [reply]
1. Template components is not object oriented as I haven't been able to spend time on this yet. So you should following the official WhatsApp doc for templates: https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-message-templates
2. `MediaObjectID` is the ID of some resource pushed on Whatsapp servers. Upload a resource from this package is not yet available. You can upload images (or other resources) to WhatsApp servers using the Cloud API directly: https://developers.facebook.com/docs/whatsapp/cloud-api/reference/media#upload-media ... `LinkID` is a objet that represent the URL of some resource uploaded on somewhere.
3. During the month of July I will be enjoying a well-deserved holidays and I will have time to add improvements.
